llvm/musl: `error: no matching function for call to 'erase_if'`

`xdg-desktop-portal-hyprland` fails to compile on Gentoo Linux amd64 musl libc (LLVM 20 profile):

```bash
[26/35] /usr/lib/ccache/bin/clang++ -DXDPH_VERSION=\"1.3.9\" -I/var/tmp/portage/gui-libs/xdg-desktop-portal-hyprland-9999/work/xdg-desktop-portal-hyprland-9999/. -I/var/tmp/portage/gui-libs/xdg-desktop-portal-hyprland-9999/work/xdg-desktop-portal-hyprland-9999/protocols -isystem /usr/include/elogind -isystem /usr/lib/libffi/include -isystem /usr/include/pipewire-0.3 -isystem /usr/include/spa-0.2 -isystem /usr/include/libdrm  -O3 -pipe -march=native -g0 -Werror=odr -Werror=conditional-type-mismatch -Werror=pointer-type-mismatch -Werror=selector-type-mismatch -Werror=strict-aliasing -Wno-implicit-function-declaration -Wno-sizeof-pointer-memaccess -flto -D_FORTIFY_SOURCE=3 -stdlib=libc++ -std=gnu++23 -Wall -Wextra -Wno-unused-parameter -Wno-unused-value -Wno-missing-field-initializers -Wno-narrowing -Wno-pointer-arith -Wno-address-of-temporary -D_REENTRANT -MD -MT CMakeFiles/xdg-desktop-portal-hyprland.dir/src/shared/ScreencopyShared.cpp.o -MF CMakeFiles/xdg-desktop-portal-hyprland.dir/src/shared/ScreencopyShared.cpp.o.d -o CMakeFiles/xdg-desktop-portal-hyprland.dir/src/shared/ScreencopyShared.cpp.o -c /var/tmp/portage/gui-libs/xdg-desktop-portal-hyprland-9999/work/xdg-desktop-portal-hyprland-9999/src/shared/ScreencopyShared.cpp
samu: job failed with status 1: /usr/lib/ccache/bin/clang++ -DXDPH_VERSION=\"1.3.9\" -I/var/tmp/portage/gui-libs/xdg-desktop-portal-hyprland-9999/work/xdg-desktop-portal-hyprland-9999/. -I/var/tmp/portage/gui-libs/xdg-desktop-portal-hyprland-9999/work/xdg-desktop-portal-hyprland-9999/protocols -isystem /usr/include/elogind -isystem /usr/lib/libffi/include -isystem /usr/include/pipewire-0.3 -isystem /usr/include/spa-0.2 -isystem /usr/include/libdrm  -O3 -pipe -march=native -g0 -Werror=odr -Werror=conditional-type-mismatch -Werror=pointer-type-mismatch -Werror=selector-type-mismatch -Werror=strict-aliasing -Wno-implicit-function-declaration -Wno-sizeof-pointer-memaccess -flto -D_FORTIFY_SOURCE=3 -stdlib=libc++ -std=gnu++23 -Wall -Wextra -Wno-unused-parameter -Wno-unused-value -Wno-missing-field-initializers -Wno-narrowing -Wno-pointer-arith -Wno-address-of-temporary -D_REENTRANT -MD -MT CMakeFiles/xdg-desktop-portal-hyprland.dir/src/shared/ToplevelMappingManager.cpp.o -MF CMakeFiles/xdg-desktop-portal-hyprland.dir/src/shared/ToplevelMappingManager.cpp.o.d -o CMakeFiles/xdg-desktop-portal-hyprland.dir/src/shared/ToplevelMappingManager.cpp.o -c /var/tmp/portage/gui-libs/xdg-desktop-portal-hyprland-9999/work/xdg-desktop-portal-hyprland-9999/src/shared/ToplevelMappingManager.cpp
../xdg-desktop-portal-hyprland-9999/src/shared/ToplevelMappingManager.cpp:21:9: error: no matching function for call to 'erase_if'
   21 |         std::erase_if(m_vHandles, [&](const auto& other) { return other.get() == h; });
      |         ^~~~~~~~~~~~~
/usr/include/c++/v1/unordered_map:1809:1: note: candidate template ignored: could not match 'unordered_map' against 'std::vector'
 1809 | erase_if(unordered_map<_Key, _Tp, _Hash, _Pred, _Alloc>& __c, _Predicate __pred) {
      | ^
/usr/include/c++/v1/unordered_map:2505:1: note: candidate template ignored: could not match 'unordered_multimap' against 'std::vector'
 2505 | erase_if(unordered_multimap<_Key, _Tp, _Hash, _Pred, _Alloc>& __c, _Predicate __pred) {
      | ^
/usr/include/c++/v1/string:3960:1: note: candidate template ignored: could not match 'basic_string' against 'std::vector'
 3960 | erase_if(basic_string<_CharT, _Traits, _Allocator>& __str, _Predicate __pred) {
      | ^
../xdg-desktop-portal-hyprland-9999/src/shared/ToplevelMappingManager.cpp:26:9: error: no matching function for call to 'erase_if'
   26 |         std::erase_if(m_vHandles, [&](const auto& other) { return other.get() == h; });
      |         ^~~~~~~~~~~~~
/usr/include/c++/v1/unordered_map:1809:1: note: candidate template ignored: could not match 'unordered_map' against 'std::vector'
 1809 | erase_if(unordered_map<_Key, _Tp, _Hash, _Pred, _Alloc>& __c, _Predicate __pred) {
      | ^
/usr/include/c++/v1/unordered_map:2505:1: note: candidate template ignored: could not match 'unordered_multimap' against 'std::vector'
 2505 | erase_if(unordered_multimap<_Key, _Tp, _Hash, _Pred, _Alloc>& __c, _Predicate __pred) {
      | ^
/usr/include/c++/v1/string:3960:1: note: candidate template ignored: could not match 'basic_string' against 'std::vector'
 3960 | erase_if(basic_string<_CharT, _Traits, _Allocator>& __str, _Predicate __pred) {
      | ^
2 errors generated.
samu: subcommand failed
```

The following patch fixes the build error (see https://github.com/miracle-wm-org/miracle-wm/issues/460 for reference):
```patch
diff --git a/src/shared/ToplevelMappingManager.cpp b/src/shared/ToplevelMappingManager.cpp
index 9fb93fc..cb5c6db 100644
--- a/src/shared/ToplevelMappingManager.cpp
+++ b/src/shared/ToplevelMappingManager.cpp
@@ -1,5 +1,6 @@
 #include "ToplevelMappingManager.hpp"
 #include "../helpers/Log.hpp"
+#include <vector>
 
 CToplevelMappingManager::CToplevelMappingManager(SP<CCHyprlandToplevelMappingManagerV1> mgr) : m_pManager(mgr) {
     Debug::log(LOG, "[toplevel mapping] registered manager");

```

--- a/src/shared/ToplevelMappingManager.cpp
+++ b/src/shared/ToplevelMappingManager.cpp
@@ -1,5 +1,6 @@
 #include "ToplevelMappingManager.hpp"
 #include "../helpers/Log.hpp"
+#include <vector>
 
 CToplevelMappingManager::CToplevelMappingManager(SP<CCHyprlandToplevelMappingManagerV1> mgr) : m_pManager(mgr) {
     Debug::log(LOG, "[toplevel mapping] registered manager");

